worlds/civ_6/__init__.py

```diff
@@ -0,0 +1,192 @@
+    location_name_to_id = {
+        location.name: location.code for location in generate_flat_location_table().values()}
+
+    item_table: Dict[str, CivVIItemData] = {}
+    location_by_era: Dict[EraType, Dict[str, CivVILocationData]]
+
+    data_version = 1
+    required_client_version = (0, 4, 5)
+
+    def __init__(self, multiworld: "MultiWorld", player: int):
+        super().__init__(multiworld, player)
+        self.location_by_era = generate_era_location_table()
+
+        self.location_table: Dict[str, CivVILocationData] = {}
+        self.item_table = generate_item_table()
+
+        for _era, locations in self.location_by_era.items():
+            for _item_name, location in locations.items():
+                self.location_table[location.name] = location
+
+    def get_filler_item_name(self) -> str:
+        return get_random_filler_by_rarity(self, FillerItemRarity.COMMON, self.item_table).name
+
+    def create_regions(self) -> None:
+        create_regions(self, self.options, self.player)
+
+    def set_rules(self) -> None:
+        if self.options.boostsanity:
+            create_boost_rules(self)
+
+    def create_item(self, name: str) -> Item:
+        item: CivVIItemData = self.item_table[name]
+        classification = item.classification
+        if self.options.boostsanity:
+            if item.civ_name in BOOSTSANITY_PROGRESSION_ITEMS:
+                classification = ItemClassification.progression
+
+        return CivVIItem(item, self.player, classification)
+
+    def create_items(self) -> None:
+        progressive_era_item = None
+        for item_name, data in self.item_table.items():
+            # Don't add progressive items to the itempool here
+            if data.item_type == CivVICheckType.PROGRESSIVE_DISTRICT:
+                continue
+            if data.item_type == CivVICheckType.ERA:
+                # Don't add era items in this way
+                progressive_era_item = data
+                continue
+            if data.item_type == CivVICheckType.GOODY:
+                continue
+
+                # If we're using progressive districts, we need to check if we need to create a different item instead
+            item_to_create = item_name
+            if self.options.progression_style != "none":
+                item: CivVIItemData = self.item_table[item_name]
+                if item.progression_name != None:
+                    item_to_create = self.item_table[item.progression_name].name
+
+            self.multiworld.itempool += [self.create_item(
+                item_to_create)]
+
+        # Era items
+        if self.options.progression_style == "eras_and_districts":
+            # Add one less than the total number of eras (start in ancient, don't need to find it)
+            for era in EraType:
+                if era.value == "ERA_ANCIENT":
+                    continue
+                self.multiworld.itempool += [self.create_item(
+                    progressive_era_item.name)]

```

Comment:
Alright so at the beginning of create_items, you loop through the item table. Whenever you hit an era check type, you set `progressive_era_item` equal to the data there.
Then after the loop is done, you check if the progression style is eras and districts, and if it is and it passes some checks, you create the item for `progressive_era_item.name` and add it to the item pool.

So, you seem to be adding a bunch of copies of whatever the last era item is in the item pool?

This seems like a bug, but please let me know if it isn't

Reply:
Ah I could probably make this more descriptive but the `progressive_era_item` is a singular item type. So the item_table with all the data I need looks like this:
```
Progressive Harbor: data
Progressive Era: data
Tech Archery: data
Civic Craftsmanship: data
```

So if you want to reach the 5th era in the game you'd need at least 4 `Progressive Era` items. My python understanding has evolved a little since I wrote this originally so I'll move this around so it is more clear. Nice catch 🔥

Reply:
Ok just pushed up an update for it, if you get a chance take a look and let me know if it makes more sense 👍

Reply:
Looks good to me, I have no idea how to resolve this comment